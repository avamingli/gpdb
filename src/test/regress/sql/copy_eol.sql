--
-- copy data from file with different EOL and NEWLINE option.
--
CREATE TABLE copy_eol(id int, seq text);

--- EOL: '\n'
\!echo -n -e '1\x1es1\\\n2\x1es2\\\n' > /tmp/copy_lf.file;
COPY copy_eol FROM '/tmp/copy_lf.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b', NEWLINE 'LF');
COPY copy_eol FROM '/tmp/copy_lf.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b'); 

--- EOL: '\r'
\!echo -n -e '1\x1es1\\\r2\x1es2\\\r' > /tmp/copy_cr.file;
COPY copy_eol FROM '/tmp/copy_cr.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b', NEWLINE 'CR');
COPY copy_eol FROM '/tmp/copy_cr.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b');

--- EOL: '\r\n'
\!echo -n -e '1\x1es1\\\r\n2\x1es2\\\r\n' > /tmp/copy_crlf.file;
COPY copy_eol FROM '/tmp/copy_crlf.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b', NEWLINE 'CRLF');
COPY copy_eol FROM '/tmp/copy_crlf.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b');

--- EOL: '\r\n' with a '\r' in data
\!echo -n -e '1\x1es1\\\r\n2\x1es\\\r2\\\r\r\n3\x1es3\r\n' > /tmp/copy_crlf_1.file;
COPY copy_eol FROM '/tmp/copy_crlf_1.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b', NEWLINE 'CRLF');
COPY copy_eol FROM '/tmp/copy_crlf_1.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b');

SELECT count(*) FROM copy_eol;

DROP TABLE copy_eol;
