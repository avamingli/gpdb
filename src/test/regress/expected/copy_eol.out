--
-- copy data from file with different EOL and NEWLINE option.
--
CREATE TABLE copy_eol(id int, seq text);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'id' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
--- EOL: '\n'
\!echo -n -e '1\x1es1\\\n2\x1es2\\\n' > /tmp/copy_lf.file;
COPY copy_eol FROM '/tmp/copy_lf.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b', NEWLINE 'LF');
COPY copy_eol FROM '/tmp/copy_lf.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b'); 
--- EOL: '\r'
\!echo -n -e '1\x1es1\\\r2\x1es2\\\r' > /tmp/copy_cr.file;
COPY copy_eol FROM '/tmp/copy_cr.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b', NEWLINE 'CR');
COPY copy_eol FROM '/tmp/copy_cr.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b');
--- EOL: '\r\n'
\!echo -n -e '1\x1es1\\\r\n2\x1es2\\\r\n' > /tmp/copy_crlf.file;
COPY copy_eol FROM '/tmp/copy_crlf.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b', NEWLINE 'CRLF');
COPY copy_eol FROM '/tmp/copy_crlf.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b');
--- EOL: '\r\n' with a '\r' in data
\!echo -n -e '1\x1es1\\\r\n2\x1es\\\r2\\\r\r\n3\x1es3\r\n' > /tmp/copy_crlf_1.file;
COPY copy_eol FROM '/tmp/copy_crlf_1.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b', NEWLINE 'CRLF');
COPY copy_eol FROM '/tmp/copy_crlf_1.file' (FORMAT 'text', DELIMITER E'\x1e' , NULL E'\x1b\x4e',  ESCAPE E'\x1b');
SELECT count(*) FROM copy_eol;
 count 
-------
    18
(1 row)

DROP TABLE copy_eol;
